wrappers/services.go

```diff
@@ -1566,19 +1566,34 @@ type RestartServicesFlags struct {
 func RestartServices(svcs []*snap.AppInfo, flags *RestartServicesFlags, inter interacter, tm timings.Measurer) error {
 	sysd := systemd.New(systemd.SystemMode, inter)
 
+	unitNames := make([]string, 0, len(svcs))
 	for _, srv := range svcs {
 		// they're *supposed* to be all services, but checking doesn't hurt
 		if !srv.IsService() {
 			continue
 		}
+		unitNames = append(unitNames, srv.ServiceName())
+	}
+
+	unitStatuses, err := sysd.Status(unitNames...)
+	if err != nil {
+		return err
+	}
+
+	for _, unit := range unitStatuses {
+		// restart all active services (even if disabled), as described in
+		// https://forum.snapcraft.io/t/command-line-interface-to-manipulate-services/262/47
```

Comment:
Thanks for adding this comment!

Reply:
The logic here seems fine, but now reading my past comment, I am reminded of my original question to @pedronis, I think it's fine to leave this PR as-is, but I'm still curious about the case of oneshot or failed services, if those should be restarted as well when you run `snap restart`, that still seems logical to me

Reply:
This is more complicated, as far I can tell this changes the behavior of   "snap restart snap.svc1 snap.svc2" as well, no? which wasn't the intention. The comments are strictly about "snap restart snap".